packages/grid/x-data-grid/src/components/panel/GridPanel.tsx

```diff
@@ -98,12 +97,12 @@ const GridPanel = React.forwardRef<HTMLDivElement, GridPanelProps>((props, ref)
   const [anchorEl, setAnchorEl] = React.useState<Element | null>(null);
 
   React.useEffect(() => {
-    const columnHeadersElement = apiRef.current.rootElementRef?.current?.querySelector(
-      `.${gridClasses.columnHeaders}`,
+    const panelAnchor = apiRef.current.rootElementRef?.current?.querySelector(
+      '[data-id="gridPanelAnchor"]',
```

Comment:
Fixes this:

https://github.com/mui/mui-x/assets/13808724/62b36b90-7199-4090-b60e-d403a72b1430

Reply:
Discussion continued a bit in https://github.com/mui/mui-x/pull/15022#issuecomment-2424213768.